### DISCUSSION TITLE
A question about custom language support

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#187] [General]
If a language, for example, is defined in both `eepp` source code and `Adept.json`, then which takes precedent?

**Comment by SpartanJ** [comment]
It always prioritizes any loaded language, so it basically will try to use `Adept.json`. The idea is to let the user do their own modifications, also you can export the language definition to the json format to extend it (I guess you know since you mentioned the json file) with the `export-lang` CLI option:
`ecode --export-lang=cpp --export-lang-path=./cpp.json`

**Reply by ghost** [reply]
How to export all of the languages?

**Reply by SpartanJ** [reply]
You can achieve it with some script by getting the language names from the `ecode --health` output, this should work on Linux and macOS:

```bash
#!/bin/bash
ecode --health --health-format=ascii | tail -n +2 | while read -r line; do
	lang_col="${line%%✓*}"
	lang=$(echo "$lang_col" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')  # Trim whitespace
	ecode --export-lang="$lang" --export-lang-path=./langs/"$lang".json
done
```
